src/modules/mc_hover_thrust_estimator/MulticopterHoverThrustEstimator.cpp

```diff
@@ -105,76 +113,104 @@ void MulticopterHoverThrustEstimator::Run()
 
 	perf_begin(_cycle_perf);
 
-	vehicle_land_detected_s vehicle_land_detected;
-	vehicle_status_s vehicle_status;
-	vehicle_local_position_s local_pos;
+	if (_vehicle_land_detected_sub.updated()) {
+		vehicle_land_detected_s vehicle_land_detected;
 
-	if (_vehicle_land_detected_sub.update(&vehicle_land_detected)) {
-		_landed = vehicle_land_detected.landed;
+		if (_vehicle_land_detected_sub.copy(&vehicle_land_detected)) {
+			_landed = vehicle_land_detected.landed;
+		}
 	}
 
-	if (_vehicle_status_sub.update(&vehicle_status)) {
-		_armed = (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED);
+	if (_vehicle_status_sub.updated()) {
+		vehicle_status_s vehicle_status;
+
+		if (_vehicle_status_sub.copy(&vehicle_status)) {
+			_armed = (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED);
+		}
 	}
 
-	if (_vehicle_local_pos_sub.update(&local_pos)) {
+	// always copy latest position estimate
+	vehicle_local_position_s local_pos{};
```

Comment:
```suggestion
	vehicle_local_position_s local_pos;
```

Reply:
It's intentional here so that the timestamp check below can't fail.

Reply:
Above we return in the case that local position is not updated, so I am not sure if it is strictly necessary. But better to be safe.